pytorch_lightning/utilities/enums.py

```diff
@@ -55,12 +58,12 @@ def __getattribute__(cls, name: str) -> Any:
         return obj
 
     def __getitem__(cls, name: str) -> Any:
-        member = super().__getitem__(name)
+        member: _OnAccessEnumMeta = super().__getitem__(name)
```

Comment:
This is likely not working. Can you use a class for typing that is not yet fully specified?

Maybe add typing with `Self/"_OnAccessEnumMeta"`